### DISCUSSION TITLE
What is Vortice.Win32?

### DISCUSSION THREAD

**Discussion by Odepax** [amerkoleci/Vortice.Windows#392] [General]
The [Vortice.Win32](https://github.com/amerkoleci/Vortice.Win32/)'s README states that:

> [It] is [...] low-level bindings for Win32, generation is based on json files from [win32json](https://github.com/marlersoft/win32json), follows naming conventions of Vortice.Windows but small GC allocation and using unsafe struct for COM objects.

Is `Vortice.Win32` envisioned to replace `Vortice.Windows` and `Vortice.Mathematics` for DirectX bindings (whether in a near or far future)?\
If not, what use case(s) differentiate `Vortice.Win32` from `Vortice.Windows`?

**Comment by amerkoleci** [comment]
Vortice.Win32 aims to have low level (almost 0 GC support), so there is a lot of unsafe and struct usage, inheritance doesn't actually work as Vortice.Windows (where we use classes and marshalling is handled for end user), using Vortice.Win32 makes the user to take care of marshalling stuff, cast between structs and unsafe code usage.

**Reply by Odepax** [reply]
Why/In what situation would you use `Vortice.Win32` v.s. `Vortice.Windows`?

**Reply by amerkoleci** [reply]
Where performance really matter, if you want to avoid the unsafe/low level part of .NET then **Vortice.Windows** is the way to go, if you want control, take care of marshal by yourself and want to really avoid GC and allocation then **Vortice.Win32**

I see it like levels, beginner/intermediate can use Vortice.Windows and just don't bother, advanced users can use Vortice.Win32 and fully control their stuff.

**Reply by Odepax** [reply]
Will `Vortice.Windows` dogfood on `Vortice.Win32`?
